CondFormats/JetMETObjects/interface/MEtXYcorrectParameters.h

```diff
@@ -0,0 +1,183 @@
+      : mDefinitions(fDefinitions),mRecords(fRecords) { valid_ = true;}
+    //-------- Member functions ----------
+    const Record& record(unsigned fBin)                          const {return mRecords[fBin]; }
+    const Definitions& definitions()                             const {return mDefinitions;   }
+    unsigned size()                                              const {return mRecords.size();}
+    unsigned size(unsigned fVar)                                 const;
+    int binIndex(const std::vector<float>& fX)                   const;
+    int neighbourBin(unsigned fIndex, unsigned fVar, bool fNext) const;
+    std::vector<float> binCenters(unsigned fVar)                 const;
+    void printScreen()                                           const;
+    void printScreen(const std::string& Section)                 const;
+    void printFile(const std::string& fFileName)                 const;
+    void printFile(const std::string& fFileName, const std::string& Section)const;
+    bool isValid() const { return valid_; }
+
+  private:
+    //-------- Member variables ----------
+    MEtXYcorrectParameters::Definitions         mDefinitions;
+    std::vector<MEtXYcorrectParameters::Record> mRecords;
+    bool                                        valid_; /// is this a valid set?
+
+  COND_SERIALIZABLE;
+};
+
+
+class MEtXYcorrectParametersCollection {
+ public:
+  enum Level_t { shiftMC=0,
+    		 shiftDY=1,
+    		 shiftTTJets=2,
+    		 shiftWJets=3,
+    		 shiftData=4,
+		 N_LEVELS=5
+  };
+
+  typedef int                            key_type;
+  typedef std::string                    label_type;
+  typedef MEtXYcorrectParameters         value_type;
+  typedef std::pair<key_type,value_type> pair_type;
+  typedef std::vector<pair_type>         collection_type;
+
+  // Constructor... initialize all three vectors to zero
+  MEtXYcorrectParametersCollection() {
+    correctionsShift_.clear();
+  }
+
+  // Add a MEtXYshiftParameter object, for each source 
+  void push_back( key_type i, value_type const & j, label_type const & flav = "" );
+
+  // Access the MEtXYshiftParameter via the key k.
+  // key_type is hashed to deal with the three collections
+  MEtXYcorrectParameters const & operator[]( key_type k ) const;
+
+  // Access the MEtXYshiftParameter via a string. 
+  // Will find the hashed value for the label, and call via that 
+  // operator. 
+  MEtXYcorrectParameters const & operator[]( std::string const & label ) const {
+    return operator[]( findKey(label) );
+  }
+
+  // Get a list of valid keys. These will contain hashed keys
+  // that are aware of all three collections. 
+  void validKeys(std::vector<key_type> & keys ) const;
+
+
+  // Helper method to find all of the sections in a given 
+  // parameters file
+  static void getSections( std::string inputFile,
+			   std::vector<std::string> & outputs );
+
+  static key_type getShiftMcFlavBin( std::string const & Flav );
```

Comment:
Hi @d4space - why are these static? they reference member data, so they should not be